src/index.ts

```diff
@@ -298,10 +305,27 @@ export function apply(ctx: Context, config: Config) {
         ? session.text('.pending', [globalTasks.size])
         : session.text('.waiting'))
 
+      if (config.globalConcurrency) {
+        if (globalTasks.size >= config.globalConcurrency) {
+          const pendingId = container.pop()
+          globalPending.add(pendingId)
+          await new Promise<void>((resolve) => ctx.once('novelai/finish', (id) => {
+            if (id === pendingId) {
+              resolve()
+            }
+          }))
```

Comment:
`ctx.once()` will only be called once. If in that time, `id` is not equal to `pendingId`, the promise will never be resolved.

```suggestion
          await new Promise<void>((resolve) => {
            const dispose = ctx.on('novelai/finish', (id) => {
              if (id === pendingId) {
                resolve()
                dispose()
              }
            }
          }))
```

Reply:
Make sense.